### DISCUSSION TITLE
How to override the `die` action?

### DISCUSSION THREAD

**Discussion by minhduc0711** [gama-platform/gama#97] [Q&A] [answered]
I want to clean up some states of an agent before it is killed, and here's what I have in driving skill:
```
@action(
	name = "die",
	doc = @doc(
		value = "remove the driving agent from its current road and make it die",
		examples = { @example("do die") }
	)
)
public void primDieWrapper(final IScope scope) throws GamaRuntimeException {
	AbstractAgent vehicle = (AbstractAgent) getCurrentAgent(scope);
	if (!vehicle.dead() && getCurrentRoad(vehicle) != null) {
		unregister(scope, vehicle);
	}
	vehicle.primDie(scope);
}
```
But this method is not invoked when `do die` is used in GAML.

Is there a proper way to do this?

**Comment by AlexisDrogoul** [comment, marked as answer]
H, there are no real ways to achieve this. What you can do, however is either posting some actions at the simulation level that will be executed every step (for example to verify the registration of vehicles to roads) or add a population listener to the population of vehicles the first time the skill is registered to react when one specific agent is killed.
